docs/guides/flax_on_pjit.md

```diff
@@ -0,0 +1,437 @@
+
+```python
+class DotReluDot(nn.Module):
+  depth: int
+  @nn.compact
+  def __call__(self, x):
+    W1 = self.param(
+        'W1', 
+        nn.with_partitioning(nn.initializers.xavier_normal(), (None, 'model')),
+        (x.shape[-1], self.depth))
+
+    y = jax.nn.relu(jnp.dot(x, W1))
+    # force a local sharding annotation.
+    y = with_sharding_constraint(y, PartitionSpec('data', 'model'))
+
+    W2 = self.param(
+        'W2', 
+        nn.with_partitioning(nn.initializers.xavier_normal(), ('model', None)),
+        (self.depth, x.shape[-1]))
+
+    z = jnp.dot(y, W2)
+    # force a local sharding annotation.
+    z = with_sharding_constraint(z, PartitionSpec('data', None))
+
+    # Return a tuple to conform with the API `nn.scan` as shown in the cell below.
+    return z, None
+
+```
+
+Note that values like `'data'`, `'model'` or `None` are passed into these API calls. This refers to how each dimension of this data should be sharded - either across one of the device mesh dimensions, or not sharded at all.
+
+For example:
+
+* When we define `W1` with shape `(x.shape[-1], self.depth)` and annotate as `(None, 'model')`):
+
+  * The first dimension (of length `x.shape[-1]`) will be replicated across all devices.
+ 
+  * The second dimension (of length `self.depth`) will be sharded over the `model` axis of the device mesh. This means `W1` will be sharded 4-way on devices `(0, 4)`, `(1, 5)`, `(2, 6)` and `(3, 7)`, on this dimension.
+  
+* When we annotate the output `z` as `('data', None)`:
+
+  * The first dimension (aka. the batch dimension) will be sharded over the `data` axis, which  means half of the batch will be processed on devices `0-3` and another half on devices `4-7`.
+  
+  * The second dimension (aka. the data depth dimension) will be replicated across all devices.
+
+
+## Lifted transforms
+
+Now define the `MLP` model as multiple layers of `DotReluDot`.
+
+To replicate identical layers, you can either use `flax.linen.scan` or a for-loop: `flax.linen.scan` can offer faster compilation times, whereas the for-loop can be faster on runtime. This guide uses `flax.linen.scan` simply to show that [Flax lifted transforms](https://flax.readthedocs.io/en/latest/advanced_topics/lift.html#supported-transformations) work together with [JAX `pjit`](https://jax.readthedocs.io/en/latest/jax.experimental.pjit.html).
+
+**Note:** `flax.linen.scan` will introduce another dimension for the params (the dimension over which `scan` is applied), and you need to use the `metadata_params` argument to annotate the partition of this dimension. Since the parameters inside your `DotReluDot` (a sub-`Module`) are already sharded along the `model` axis, you don't need to partition multiple layers across the `model` dimension here, and therefore you should denote it as `None`.
```

Comment:
this "note" is formatted differently from other "notes" - is this on purpose?
![image](https://user-images.githubusercontent.com/16050001/209135442-fb5202c5-26ab-4cff-ae82-1b2d7b99e63c.png)
vs
![image](https://user-images.githubusercontent.com/16050001/209135488-7e447e95-5e56-4154-a772-8b3e0fc91518.png)

Reply:
@andsteing We are doing a 2nd (3rd?) review with @IvyZX asap, going through the second half. Will make sure "Note:" is formatted consistently :+1: Thanks for the feedback!!